homeassistant/components/group/sensor.py

```diff
@@ -0,0 +1,405 @@
+        self._attr_name = name
+        if name == DEFAULT_NAME:
+            self._attr_name = f"{DEFAULT_NAME} {sensor_type}".capitalize()
+        self._attr_extra_state_attributes = {ATTR_ENTITY_ID: entity_ids}
+        self._attr_unique_id = unique_id
+        self.mode = all if mode else any
+        self._sensor_attr = SENSOR_TYPE_TO_ATTR[self._sensor_type]
+        self.min_value: float | None = None
+        self.max_value: float | None = None
+        self.mean: float | None = None
+        self.last: float | None = None
+        self.median: float | None = None
+        self.range: float | None = None
+        self.sum: float | None = None
+        self.min_entity_id: str | None = None
+        self.max_entity_id: str | None = None
+        self.last_entity_id: str | None = None
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+
+        @callback
+        def async_state_changed_listener(event: Event) -> None:
+            """Handle child updates."""
+            self.async_set_context(event.context)
+            self.async_defer_or_update_ha_state()
+
+        self.async_on_remove(
+            async_track_state_change_event(
+                self.hass, self._entity_ids, async_state_changed_listener
+            )
+        )
+
+        await super().async_added_to_hass()
+
+    @callback
+    def async_update_group_state(self) -> None:
+        """Query all members and determine the sensor group state."""
+        states = []
+        test_states = []
+        last_updated = now(UTC) - timedelta(days=1)
+        sensor_values: list[tuple[str, Any]] = []
+        for entity_id in self._entity_ids:
+            if (state := self.hass.states.get(entity_id)) is not None:
+                states.append(state.state)
+                try:
+                    sensor_values.append((entity_id, float(state.state)))
+                except ValueError:
+                    test_states.append(False)
+                    _LOGGER.warning(
+                        "Unable to use state. Only numerical states are supported, entity %s with value %s excluded from calculation",
+                        entity_id,
+                        state.state,
+                    )
+                    continue
+                test_states.append(True)
+                if state.last_updated > last_updated:
+                    self.last = float(state.state)
+                    self.last_entity_id = entity_id
+
+        # Set group as unavailable if all members are unavailable or missing
+        self._attr_available = any(state != STATE_UNAVAILABLE for state in states)
+
+        valid_state = self.mode(
+            state not in (STATE_UNKNOWN, STATE_UNAVAILABLE) for state in states
+        )
+        valid_state_numeric = self.mode(numeric_state for numeric_state in test_states)
+
+        if not valid_state or not valid_state_numeric:
+            # Set as unknown if any / all member is unknown or unavailable
```

Comment:
This comment is misleading since also non-numeric states are blocked

Reply:
I still think this comment is misleading?

Reply:
Removed the comment as it adds nothing.